docs/architecture/adr-template.md

```diff
@@ -8,13 +8,47 @@
 
 > This section contains all the context one needs to understand the current state, and why there is a problem. It should be as succinct as possible and introduce the high level idea behind the solution.
 
+## Alternative Approaches
+
+> This section contains information around alternative options that are considered before making a decision. It should contain a explanation on why the alternative approach(es) were not chosen.
+
 ## Decision
 
 > This section explains all of the details of the proposed solution, including implementation details.
 > It should also describe affects / corollary items that may need to be changed as a part of this.
 > If the proposed change will be large, please also indicate a way to do the change to maximize ease of review.
 > (e.g. the optimal split of things to do between separate PR's)
 
+## Detailed Design
+
+> This section does not need to be filled in at the start of the ADR, but must be completed prior to the merging of the implementation.
+
+Here are some common questions that get answered as part of the detailed design:
+
+- What are the user requirements?
+
+- What systems will be affected?
+
+- What new data structures are needed, what data structures will be changed?
+
+- What new APIs will be needed, what APIs will be changed?
+
+- What are the efficiency considerations (time/space)?
+
+- What are the expected access patterns (load/throughput)?
+
+- Are there any logging, monitoring or observability needs?
+
+- Are there any security considerations?
+
+- Are there any privacy considerations?
+
+- How will the changes be tested?
+
+- Will these changes require a breaking (major) release?
+
+- Does this change require coordination with the SDK or other?
+
 ## Status
 
 > A decision may be "proposed" if it hasn't been agreed upon yet, or "accepted" once it is agreed upon. If a later ADR changes or reverses a decision, it may be marked as "deprecated" or "superseded" with a reference to its replacement.
```

Comment:
Thinking out loud: Do we want to add another status "implemented" since there could be quite a discrepancy between when and ADR is accepted and when the code base actually reflects this

Reply:
I'd be in favor

Reply:
good call!